x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedUpdate.java

```diff
@@ -117,8 +118,17 @@ public DatafeedUpdate(StreamInput in) throws IOException {
                 in.readStringList();
             }
         }
-        this.query = in.readOptionalNamedWriteable(QueryBuilder.class);
-        this.aggregations = in.readOptionalWriteable(AggregatorFactories.Builder::new);
+        if (in.getVersion().before(Version.V_7_1_0)) {
+            this.query = QUERY_TRANSFORMER.toMap(in.readNamedWriteable(QueryBuilder.class));
```

Comment:
Pre 7.1 was using `in.readOptionalNamedWriteable(QueryBuilder.class)`, so it needs to use that same function here too, even if we now think the "Optional" was not necessary.

Reply:
yep, daggum, Intellj autocomplete fail. Should keep it consistent even if it was never `null`